### PR TITLE
Rename k8s-local-volume-provisioner to local-csi-driver

### PR DESCRIPTION
**Pull request by ylebi**
This PR fixes https://github.com/scylladb/local-csi-driver/issues/39

Rename `k8s-local-volume-provisioner` to `local-csi-driver`

### CODE REVIEW

**Comment by tnozicka** [comment]
Tide didn't survive, we'll need to update CI configs first. We'd have needed it for the image dir too to pass the CI.

File: pkg/cmd/local-csi-driver/driver.go [hunk 1]
```diff
@@ -9,15 +9,15 @@ import (
...
-	"github.com/scylladb/k8s-local-volume-provisioner/pkg/genericclioptions"
-	"github.com/scylladb/k8s-local-volume-provisioner/pkg/signals"
-	"github.com/scylladb/k8s-local-volume-provisioner/pkg/util/fs"
-	"github.com/scylladb/k8s-local-volume-provisioner/pkg/version"
+	"github.com/scylladb/local-csi-driver/pkg/cmdutil"
+	"github.com/scylladb/local-csi-driver/pkg/driver/limit"
+	"github.com/scylladb/local-csi-driver/pkg/driver/volume"
+	"github.com/scylladb/local-csi-driver/pkg/signals"
+	"github.com/scylladb/local-csi-driver/pkg/version"
+	"github.com/scylladb/local-csi-driver
```
Comment (tnozicka) [inline pkg/cmd/local-csi-driver/driver.go:17]:
```suggestion
	"github.com/scylladb/local-csi-driver"
```

Comment (tnozicka) [inline pkg/cmd/local-csi-driver/driver.go:17]:
```suggestion
	"github.com/scylladb/local-csi-driver/driver"
```

File: pkg/cmd/local-csi-driver/driver.go [hunk 2]
```diff
@@ -9,15 +9,15 @@ import (
...
-	"github.com/scylladb/k8s-local-volume-provisioner/pkg/cmdutil"
-	"github.com/scylladb/k8s-local-volume-provisioner/pkg/driver"
-	"github.com/scylladb/k8s-local-volume-provisioner/pkg/driver/limit"
-	"github.com/scylladb/k8s-local-volume-provisioner/pkg/driver/limit/xfs"
-	"github.com/scylladb/k8s-local-volume-provisioner/pkg/driver/volume"
-	"github.com/scylladb/k8s-local-volume-provisioner/pkg/genericclioptions"
-	"github.com/scylladb/k8s-local-volume-provisioner/pkg/signals"
-	"github.com/scylladb/k8s-local-volume-provisioner/pkg/util/fs"
-	"github.com/scylladb/k8s-local-volume-provisioner/pkg/version"
+	"github.com/scylladb/local-csi-driver/driver"
```
Comment (tnozicka) [inline pkg/cmd/local-csi-driver/driver.go:12]: I forgot the `pkg/` part when pointing to the unterminated string
```suggestion
	"github.com/scylladb/local-csi-driver/pkg/driver"
```

**Comment by tnozicka** [comment]
> /retest

@ylebi fyi, every push triggers the retest on its own - I only made the comments initially because this PR was created in the interim state where it got broken by the rename

File: README.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
-[![GitHub release](https://img.shields.io/github/tag/scylladb/k8s-local-volume-provisioner.svg?label=release)](https://github.com/scylladb/k8s-local-volume-provisioner/releases)
-[![Go](https://github.com/scylladb/k8s-local-volume-provisioner/actions/workflows/build-test.yaml/badge.svg?branch=master)](https://github.com/scylladb/k8s-local-volume-provisioner/actions/workflows/build-test.yaml?query=branch%3Amaster)
-[![Go Report Card](https://goreportcard.com/badge/github.com/scylladb/k8s-local-volume-provisioner)](https://goreportcard.com/report/github.com/scylladb/k8s-local-volume-provisioner)
+[![GitHub release](https://img.shields.io/github/tag/scylladb/local-csi-driver.svg?label=release)](https://github.com/scylladb/k8s-local-volocal-csi-driver
```
Comment (tnozicka) [inline README.md:1]: looks like the sed failed

File: test/e2e/set/localdriver/upstream.go [hunk 4]
```diff
@@ -6,10 +6,10 @@ import (
 	"context"
 
 	g "github.com/onsi/ginkgo/v2"
-	"github.com/scylladb/k8s-local-volume-provisioner/pkg/driver"
+	"github.com/scylladb/local-csi-driver/pkg/driver"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/api/storage/v1"
 	storagev1 "k8s.io/api/storage/v1"
+	v1 "k8s.io/api/storage/v1"
```
Comment (tnozicka) [inline test/e2e/set/localdriver/upstream.go:12]: aliasing `v1` with `v1` isn't needed
(this line shouldn't need touching based on the previous value https://github.com/scylladb/local-csi-driver/pull/56/files#diff-09b8c92c3407ce15738d1d7a1c08f6a4a8f71f1bd9ecd397d4037230c9b2e98dL11)
```suggestion
	"k8s.io/api/storage/v1"
```

**Comment by tnozicka** [comment]
(also please squash the 5 commits)

**Comment by tnozicka** [comment]
https://github.com/scylladb/local-csi-driver/issues/49#issuecomment-2230280219
/retest

File: README.md (hunk 3 above)
Comment (ylebi) [reply README.md]: fixed. https://github.com/scylladb/local-csi-driver/pull/56/commits/f90bb3311af511392b80a4f758cf563de89e30de

**Comment by tnozicka** [comment]
the changes now lgtm

> https://github.com/scylladb/local-csi-driver/pull/56#issuecomment-2228565415

There is 8 commits in the PR. Please squash the commits and you are good to go. E.g. 1 commit `Rename k8s-local-volume-provisioner to local-csi-driver` would be appropriate. Thanks.

**Comment by ylebi** [comment]
https://prow.scylla-operator.scylladb.com/view/gs/scylla-operator-prow/pr-logs/pull/scylladb_local-csi-driver/56/pull-local-csi-driver-master-sanity/1813212762289999872#1:build-log.txt%3A22
/retest

**Comment by tnozicka** [comment]
thanks for the updates

/approve
/lgtm

**Pull request merged**